RingPublishingTracking/src/main/java/com/ringpublishing/tracking/internal/delegate/ConfigurationManager.kt

```diff
@@ -0,0 +1,99 @@
+package com.ringpublishing.tracking.internal.delegate
+
+import com.ringpublishing.tracking.data.RingPublishingTrackingConfiguration
+import com.ringpublishing.tracking.internal.config.OperationMode
+import com.ringpublishing.tracking.internal.data.UserData
+import com.ringpublishing.tracking.internal.log.Logger
+import java.net.URL
+
+internal class ConfigurationManager
+{
+
+	internal lateinit var ringPublishingTrackingConfiguration: RingPublishingTrackingConfiguration
+
+	private val operationMode = OperationMode()
+
+	private val userData = UserData()
+
+	private var currentAdvertisementArea: String? = null
+
+	var currentReferrer: URL? = null
+
+	var currentPublicationUrl: URL? = null
+
+	private var currentStructurePath = mutableListOf<String>()
+
+	var currentIsPartialView = false
+
+	var primaryId: String? = null
+
+	var secondaryId: String? = null
+
+	fun initializeConfiguration(ringPublishingTrackingConfiguration: RingPublishingTrackingConfiguration)
+	{
+		this.ringPublishingTrackingConfiguration = ringPublishingTrackingConfiguration
```

Comment:
jakbys tu sobie przypisał do zmiennych np. currentAdvertisementArea to nie musisz tych ifów robic nizej